### PR TITLE
Add support for config files with `.php` extension

### LINKED ISSUE #246: Support config files with `.php` extension
In addition to the already supported `.twig_cs` and `.twig_cs.dist` files, I suggest to add the support for `.twig_cs.php` and `.twig_cs.dist.php`
Inspired from PHP-CS-Fixer, they already did that in https://github.com/FriendsOfPHP/PHP-CS-Fixer/pull/5607, they also have removed the support of old names in https://github.com/FriendsOfPHP/PHP-CS-Fixer/pull/5609

Benefits of this move are:
- IDE will automatically detect the file as being a PHP file
- the config file can be targeted in a generic way with `*.php` like other PHP files to be included in other CS tools like PHP-CS-Fixer, PHP_CodeSniffer, etc, no need to maintain it as a particular case

I can try to come with a PR if the idea is accepted

### PR DESCRIPTION
**Pull request by yguedidi**
Fixes #246

Better reviewed commit by commit

### CODE REVIEW

**Review by Ciloe — COMMENTED** [review]
Seems good, but check tests ?

**Comment by OwlyCode** [comment]
Looks good to me! Can we keep the test on the old convention until we release the BC break to remove it?

**Comment by localheinz** [comment]
@Ciloe @OwlyCode @yguedidi 

How do you feel about merging #248 first, then we can rebase this branch and adjust the tests?

**Comment by yguedidi** [comment]
@localheinz looks like a good plan! I should have added tests for the ConfigResolver in the first place :sweat_smile:

**Review by localheinz — APPROVED** [review]
👍

**Comment by yguedidi** [comment]
thank you @localheinz for the update!
one additional change that could be done is to reverse `.twig_cs.dist.php` and `.twig_cs`, so that local config always win over dist config

**Comment by localheinz** [comment]
@yguedidi 

Sounds good, let me adjust!

File: README.md [hunk 1]
```diff
@@ -140,15 +140,22 @@ you must use the `--config` option:
...
+twigcs --config ~/dirB/.twig_cs.dist.php # Will lint templates in ~/dirA with the config of ~/dirB
 ```
 
-By default, the files `.twig_cs` and `.twig_cs.dist` are looked up in your current working directory (CWD).
+By default, the files
+
+- `.twig_cs.php`
+- `.twig_cs`
+- `.twig_cs.dist.php`
+- `.twig_cs.dist`
```
Comment (localheinz) [inline README.md:151]: @yguedidi 

I have turned this into a list, making the order perhaps a bit more clear. What do you think?

Comment (yguedidi) [reply README.md:151]: Indeed better!

**Review by Ciloe — APPROVED** [review]

**Comment by localheinz** [comment]
Thank you, @Ciloe, @OwlyCode, and @yguedidi!

**Comment by localheinz** [comment]
@OwlyCode 

Do you want to prepare a release later?

**Pull request merged**